kernel/Makefile

```diff
@@ -0,0 +1,169 @@
+### Build configuration ###
+
+# C compiler used to compile kernel C code
+# Compiler built by xbstrap is in its build directory
+CC = ../build/xbstrap-dir/tools/kernel-gcc/bin/x86_64-elf-gcc
+
+# Assembler used to assemble kernel asm files
+AS = nasm
+
+# Linker used to generate kernel binary from object files
+# We use gcc itself for linking
+# TODO: replace with x86_64-lezione-gcc once we have xbstrap config
+LD = $(CC)
+
+# Common C flags (used both in debug and release mode)
+# -c - create object file
+# -mno-80387, -mno-mmx, -mno-sse, -mno-sse2 - do not generate code
+# that uses extended registers (87 fpu, MMX, SSE, SSE2)
+# -mno-red-zone - do not assume red zone 
+# -mcmodel=kernel - instruct gcc that kernel will be located
+# in negative 2 gigabytes of the virtual address space
+# -ffrestanding - do not assume running in standard C library
+# environment (which bare metal does not offer)
+# -fno-stack-protector
+# -Wall -Werror -Wextra - guard against simple mistakes using
+# GCC diagnostics capabilities
+# -fno-pic - do not generate position independent code. Hosted compilers
+# usually enable it by default, as they generate ASLR compatible
+# executables
+# printf library that config for printf exists
+# -nostdlib - 
+CFLAGS = -c -mno-80387 -mno-mmx -mno-sse -mno-sse2 \
```

Comment:
Specify why the no-FPU flags are necessary

Reply:
Fixed